bpfd/src/bpf.rs

```diff
@@ -555,66 +557,133 @@ impl BpfManager {
         Ok(())
     }
 
-    pub(crate) fn list_programs(&mut self) -> Result<Vec<ProgramInfo>, BpfdError> {
+    pub(crate) fn list_programs(&mut self) -> Result<Vec<ListInfo>, BpfdError> {
         debug!("BpfManager::list_programs()");
-        let programs = self
+
+        let mut bpfd_progs: HashMap<String, ProgramInfo> = self
             .programs
             .iter()
-            .map(|(id, p)| match p {
-                Program::Xdp(p) => ProgramInfo {
-                    id: *id,
-                    name: p.data.section_name.to_string(),
-                    location: p.data.location.clone(),
-                    program_type: ProgramType::Xdp as i32,
-                    attach_info: crate::command::AttachInfo::Xdp(crate::command::XdpAttachInfo {
-                        iface: p.info.if_name.to_string(),
-                        priority: p.info.metadata.priority,
-                        proceed_on: p.info.proceed_on.clone(),
-                        position: p.info.current_position.unwrap_or_default() as i32,
-                    }),
-                },
-                Program::Tracepoint(p) => ProgramInfo {
-                    id: *id,
-                    name: p.data.section_name.to_string(),
-                    location: p.data.location.clone(),
-                    program_type: ProgramType::Tracepoint as i32,
-                    attach_info: crate::command::AttachInfo::Tracepoint(
-                        crate::command::TracepointAttachInfo {
-                            tracepoint: p.info.tracepoint.to_string(),
+            .map(|(id, p)| {
+                // The key length is abbreviated to the first 16 bytes of the
+                // program name since that's what will be listed by the
+                // `aya::programs::loaded_programs()` API.
+                let prog_index = if p.section_name().len() > 16 {
+                    p.section_name()[0..16].to_string()
+                } else {
+                    p.section_name().to_string()
+                };
```

Comment:
This bit is going to need to change.... I was using prog "name" to match a program loaded by bpfd to a given `aya::ProgramInfo` but in reality that won't work, instead I'll need to use an actual unique id to do so.

Originally this was done though because the kernel limits the name to 16 bytes unfortunately see https://elixir.bootlin.com/linux/latest/source/include/uapi/linux/bpf.h#L1318

Reply:
Not relevant anymore using kernel ID as the unique ident for match a bpfd prog to it's kernel meta